src/segments/pulumi.go

```diff
@@ -0,0 +1,126 @@
+package segments
+
+import (
+	"encoding/json"
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+)
+
+type Pulumi struct {
+	props properties.Properties
+	env   platform.Environment
+
+	Stack string
+
+	Backend
+}
+
+type Backend struct {
+	URL  string
+	User string
+}
+
+type PulumiStackLSJson struct {
+	Name          string `json:"name"`
+	Current       bool   `json:"current"`
+	ResourceCount int    `json:"resourceCount"`
+	LastUpdate    string `json:"lastUpdate"`
+}
+
+type PulumiAboutJS struct {
+	Backend *PulumiBackendJS `json:"backend"`
+}
+
+type PulumiBackendJS struct {
+	URL  string `json:"url"`
+	User string `json:"user"`
+}
+
+func (p *Pulumi) Template() string {
+	return " {{if .User }}{{ .User }}@{{ end }}{{ if .URL }}{{ .URL }}{{ end }}{{if .Stack }} :: {{ .Stack }}{{ end }} "
+}
+
+func (p *Pulumi) Init(props properties.Properties, env platform.Environment) {
+	p.props = props
+	p.env = env
+}
+
+func (p *Pulumi) Enabled() bool {
+	if p.env.HasFiles("Pulumi.yaml") {
+		return p.getPulumiStackName() && p.getPulumiAbout()
+	}
+
+	return false
+}
+
+func (p *Pulumi) getPulumiAbout() bool {
+	cmd := "pulumi"
+	aboutOutput, err := p.env.RunCommand(cmd, "about", "--json")
+
+	displayError := p.props.GetBool(properties.DisplayError, false)
+	if err != nil && displayError {
+		p.setError("PULUMI ABOUT ERROR")
+		return true
+	}
+	if err != nil {
+		return false
+	}
+
+	var about PulumiAboutJS
+	err = json.Unmarshal([]byte(aboutOutput), &about)
+	if err != nil && displayError {
+		p.setError("PULUMI ABOUT DECODE ERROR, " + err.Error())
+		return true
+	}
+	if err != nil {
+		return false
+	}
+
+	p.URL = about.Backend.URL
```

Comment:
I would use the cache to store these rather than retrieve these values every time.